src/tiles/storage.rs

```diff
@@ -86,20 +86,38 @@ impl TileStorage {
         self.tiles.iter_mut()
     }
 
-    /// Remove entity at the given tile position, if there was one, leaving `None` in its place.
+    /// Removes any stored `Entity` at the given tile position, leaving `None` in its place and
+    /// returning the `Entity`.
     ///
     /// Panics if the given `tile_pos` doesn't lie within the extents of the underlying tile map.
-    pub fn remove(&mut self, tile_pos: &TilePos) {
-        self.tiles[tile_pos.to_index(&self.size)].take();
+    pub fn remove(&mut self, tile_pos: &TilePos) -> Option<Entity> {
```

Comment:
should we get rid of this entirely and make `checked_remove` the new `remove` while we're adding a return value? 0.16 is going to come with [fallible systems](https://github.com/bevyengine/bevy/pull/16589) too, and I can't actually think of a use case for panicking here. Its not chainable, so .unwrap() won't add length to chains of calls if panicing is preferred.

[upstream preference](https://github.com/bevyengine/bevy/issues/15979#issuecomment-2552997940) seems to be for returned errors, especially in cases of lesser-used apis.

Reply:
That makes sense.

The whole set of methods could really use a closer look though. Maybe I should remove the `remove` changes and we can discuss in a separate issue?

Reply:
I don't think we need to revert, but happy to discuss more generally in a separate issue. Even if we do that sort of thing the function signatures here will match and the only difference will be "no panic".